tsdb/store.go

```diff
@@ -450,35 +484,57 @@ func (s *Store) DeleteRetentionPolicy(database, name string) error {
 
 // DeleteMeasurement removes a measurement and all associated series from a database.
 func (s *Store) DeleteMeasurement(database, name string) error {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	m, err := func() (*Measurement, error) {
+		s.mu.Lock()
```

Comment:
This can be a read-lock because the indexes have their own mutexs.  It should be released after 492 too.  Taking a write lock here will lock writes and queries for all databases until this completes which is not necessary.